common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -228,7 +232,7 @@ def student_view(self, context):
                 'type': child.get_icon_class(),
                 'id': child.scope_ids.usage_id.to_deprecated_string(),
                 'bookmarked': is_bookmarked,
-                'path': " > ".join(display_names + [child.display_name or '']),
+                'path': " > ".join(display_names + [child.display_name or child.display_name_with_default]),
```

Comment:
all you need is `child.display_name_with_default`: https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/course_metadata_utils.py#L68

`display_name_with_default` will try to use `child.display_name` if it's present, and will otherwise construct a default name.

This comment applies everywhere else in this PR where you do `module.display_name or module.display_name_with_default`. You only need `module.display_name_with_default`.